samples/opengl/depthMapTriangulation/image_triangulator.cpp

```diff
@@ -0,0 +1,949 @@
+
+                resolve_dependencies(mesh, unprojected_image, Point2D(y_mid, x + step_med + step_small),
+                                     selected_vertices);  // up
+                resolve_dependencies(mesh, unprojected_image, Point2D(y_mid + step_small, x + step_med),
+                                     selected_vertices);  // left
+                resolve_dependencies(mesh, unprojected_image, Point2D(y_mid + step_small, x + step_med + step_small),
+                                     selected_vertices);  // center
+                resolve_dependencies(mesh, unprojected_image, Point2D(y_mid + step_small, x + step_med + step_med),
+                                     selected_vertices);  // right
+                resolve_dependencies(mesh, unprojected_image, Point2D(y_mid + step_med, x + step_med + step_small),
+                                     selected_vertices);  // down
+            }
+        }
+
+        // upper right
+        if (y < image_height - 1 && x_mid < image_width - 1)
+        {
+            if (y_mid >= image_height || x_right >= image_width ||
+                check_quad_error_threshold_exceeded(unprojected_image, 1, settings.RQT_error_threshold,
+                                                    Point2D(y, x_mid), Point2D(y_mid, x_mid), Point2D(y_mid, x_right),
+                                                    Point2D(y, x_right)))
+            {
+                quad_heap.push_back(std::pair<Point2D, int>(Point2D(y, x_mid), step_med + 1));
+
+                resolve_dependencies(mesh, unprojected_image, Point2D(y, x + step_med + step_small),
+                                     selected_vertices);  // up
+                resolve_dependencies(mesh, unprojected_image, Point2D(y + step_small, x + step_med),
+                                     selected_vertices);  // left
+                resolve_dependencies(mesh, unprojected_image, Point2D(y + step_small, x + step_med + step_small),
+                                     selected_vertices);  // center
+                resolve_dependencies(mesh, unprojected_image, Point2D(y + step_small, x + step_med + step_med),
+                                     selected_vertices);  // right
+                resolve_dependencies(mesh, unprojected_image, Point2D(y + step_med, x + step_med + step_small),
+                                     selected_vertices);  // down
+            }
+        }
+    }
+}
+
+
+// ----------- actual triangulation -----------
+
+void Saiga::RQT_Triangulator::triangulate_RQT_selected_vertices(
+    MyMesh& current_mesh, ImageView<OpenTriangleMesh::VertexHandle> const& selected_vertices)
+{
+    // heap for quads that need to be triangulated
+    // consists of the coordinates of the left upper corner, the side_length of the quad and a bool indicating the
+    // triangulation (false: top left to bottom right, true: top right to bottom left)
+    std::vector<std::tuple<Point2D, int, bool>> quad_heap;
```

Comment:
reserve?

Reply:
I can get an upper limit for this, but that would be way higher than needed in almost all cases. Is it still better to reserve then?